src/Processors/Transforms/AggregatingTransform.cpp

```diff
@@ -123,7 +123,10 @@ class ConvertingAggregatedToChunksWithMergingSource : public ISource
         UInt32 bucket_num = shared_data->next_bucket_to_merge.fetch_add(1);
 
         if (bucket_num >= NUM_BUCKETS)
+        {
+            data.reset();
```

Comment:
Comment is needed.

Reply:
It feels pretty self-explanatory to me here. But maybe we can write a bigger comment somewhere in .h file about this kind of cleaning and why and where we need it.